src/string-util/string-util.ts

```diff
@@ -15,50 +15,89 @@ const KOREA_LOT_NUMBER_ADDRESS_REGEXP = /읍|면|동|가/;
 );
-const EMOJI_REGEXP =
-  /(\u00a9|\u00ae|[\u2000-\u3300]|\ud83c[\ud000-\udfff]|\ud83d[\ud000-\udfff]|\ud83e[\ud000-\udfff])/gi;
+
+//이모지 정규식 표현
+const EMOJI_REGEXP = /([\uD800-\uDBFF][\uDC00-\uDFFF])/;
+//ZERO WIDTH JOINER = 다중이모지에서 이모지 끼리 붙일 때 쓰이는 코드
+const EMOJI_ZERO_WIDTH_JOINER = '\u200D';
 
 export namespace StringUtil {
   export function maskPrivacy(text: string, type: PrivacyType): string {
-    const targetMask: Record<PrivacyType, MaskingRange> = {
-      name: {
-        start: EMOJI_REGEXP.test(text.slice(0, 2)) ? 2 : 1,
-        end: EMOJI_REGEXP.test(text.slice(text.length - 2)) ? text.length - 2 : text.length - 1,
-      },
-      phone: {
-        start: /\D/.test(text) ? getMaskingIndexOfUnnormalizedPhone(text).start : 3,
-        end: /\D/.test(text) ? getMaskingIndexOfUnnormalizedPhone(text).end : text.length - 4,
-      },
+    const maskingRule: Record<PrivacyType, MaskingRange> = {
+      name: { start: 1, end: text.length - 1 },
+      phone: { start: 3, end: text.length - 4 },
       email: { start: 2, end: text.indexOf('@') },
       bankAccount: { start: 3, end: text.length - 3 },
       address: { start: getMaskingStartIndexOfAddress(text), end: text.length },
     };
 
-    const start = targetMask[type].start;
-    const end = targetMask[type].end;
+    const start = maskingRule[type].start;
+    const end = maskingRule[type].end;
     const length = type === 'name' ? end - start || 1 : end - start;
-    return getMaskedString({ text, length, maskingStart: start });
+    return getMaskedString({ text, length, maskingStart: start, type });
+
+    function getMaskedString({ text, length, maskingStart, type }: MaskingOpts): string {
+      //emoji가 있는 text에 대해서 따로 처리
+      if (EMOJI_REGEXP.test(text)) {
+        return getMaskedWithEmoji(text);
+      }
 
-    function getMaskedString({ text, length, maskingStart }: MaskingOpts): string {
       const validLength = Math.max(0, length);
       const masking = '*'.repeat(validLength);
       const textSplit = [...text];
       textSplit.splice(maskingStart, validLength, masking);
+
       return textSplit.join('');
     }
 
-    function getMaskingIndexOfUnnormalizedPhone(phone: string): { start: number; end: number } {
-      const phoneNumberParts = phone.split(/\D/);
+    function getMaskedWithEmoji(text: string): string {
+      const splitNames = splitNameByEmoji(text);
+
+      let maskedName = '';
+
+      if (splitNames.length === 2) {
+        //이름이 2글자면 맨 뒤 마스킹
+        maskedName = splitNames[0] + '*';
+      } else {
+        //맨 앞과 맨 뒤 제외 마스킹
+        for (let i = 0; i < splitNames.length; i++) {
+          if (i === 0 || i === splitNames.length - 1) {
+            maskedName += splitNames[i];
+          } else {
+            maskedName += '*';
+          }
+        }
+      }
+
+      return maskedName;
+    }
 
-      let middleIndex = Math.floor(phoneNumberParts.length / 2);
+    function splitNameByEmoji(text: string): string[] {
```

Comment:
허들상으로 들은 설명대로라면 이 함수의 결과는 글자의 array인 것이므로
String.split을 연상시키는 이 이름과 signature는 맞지 않습니다.
문자열의 글자 하나하나를 열거하는(enumerate) 거니까 이런 이름은 어떨까 합니다.
```suggestion
    function enumerateCharOfText(text: string): char[] {
```

Reply:
계속 고민하고 char가 들어가야되나 했는데 감사합니다!